fmriprep/workflows/anatomical.py

```diff
@@ -1010,33 +1041,37 @@ def init_refine_brainmask_wf(name='refine_brainmask'):
     **Outputs**
 
         out_file
-            New, refined brain mask
-
+            The selected segmentation, after resampling in native space
     """
-    workflow = pe.Workflow(name=name)
-
+    workflow = pe.Workflow(name='%s_%s' % (name, segmentation))
     inputnode = pe.Node(niu.IdentityInterface([
-        'in_file', 'ants_segs', 'subjects_dir', 'subject_id']), name='inputnode')
+        'in_file', 'subjects_dir', 'subject_id']), name='inputnode')
     outputnode = pe.Node(niu.IdentityInterface(['out_file']), name='outputnode')
-    get_aseg = pe.Node(nio.FreeSurferSource(), name='get_aseg')
+    # Extract the aseg and aparc+aseg outputs
+    fssource = pe.Node(nio.FreeSurferSource(), name='fs_datasource')
     tonative = pe.Node(fs.Label2Vol(), name='tonative')
     tonii = pe.Node(fs.MRIConvert(out_type='niigz', resample_type='nearest'), name='tonii')
-    refine = pe.Node(RefineBrainMask(), name='refine')
+
+    if segmentation.startswith('aparc'):
+        if segmentation == 'aparc_aseg':
+            def _sel(x): return x[0]
+        elif segmentation == 'aparc_a2009s':
+            def _sel(x): return x[1]
+        elif segmentation == 'aparc_dkt':
+            def _sel(x): return x[2]
+        segmentation = (segmentation, _sel)
 
     workflow.connect([
-        (inputnode, refine, [('in_file', 'in_anat'),
-                             ('ants_segs', 'in_ants')]),
-        (inputnode, get_aseg, [('subjects_dir', 'subjects_dir'),
-                               ('subject_id', 'subject_id')]),
+        (inputnode, fssource, [
+            ('subjects_dir', 'subjects_dir'),
+            ('subject_id', 'subject_id')]),
         (inputnode, tonii, [('in_file', 'reslice_like')]),
```

Comment:
I suspect this `mri_convert` approach won't work so simply if using a pre-computed FreeSurfer reconstruction with more than one input T1. This is the case that we were checking against when we ran `mri_robust_register` against the `T1.mgz` and `T1w_preproc.nii.gz` in order to get the `fsnative` transforms.

The easiest way to check this case is to get a multi-session dataset (e.g. ds114) and run a FS recon in the opposite order (`recon-all -s sub-07 -i $DS114/sub-07/ses-test/sub-07_ses-test_T1w.nii.gz -i $DS114/sub-07/ses-retest/sub-07_ses-retest_T1w.nii.gz -all`) and use that as your FreeSurfer directory. There should be a coregistration matrix that encodes more than just a reslice.

Reply:
I'll check before merging. Thanks

Reply:
Why not we just leave that edge case out? Is there a way to check if FreeSurfer's T1.mgz and T1w_preproc.nii.gz are not consistent (and just fail and recommend a wipe out)? cc @chrisfilo

Reply:
Is the edge case any dataset with more than one T1w per subject?

Reply:
The edge case is any pre-computed FreeSurfer derivatives that don't happen to align perfectly with our preprocessed T1.